src/Sentry/Protocol/SentryAttribute.cs

```diff
@@ -0,0 +1,107 @@
+namespace Sentry.Protocol;
+
+[DebuggerDisplay(@"\{ Value = {Value}, Type = {Type} \}")]
+internal readonly struct SentryAttribute
+{
+    public SentryAttribute(object value, string type)
+    {
+        Value = value;
+        Type = type;
+    }
+
+    public object Value { get; }
+    public string Type { get; }
+}
+
+internal static class SentryAttributeSerializer
+{
+    internal static void WriteAttribute(Utf8JsonWriter writer, string propertyName, SentryAttribute attribute)
+    {
+        Debug.Assert(attribute.Type is not null);
```

Comment:
do we need this assert on all calls?
it's odd to have it here but no on the others
Or only on the method these all fall into: `WriteAttributeValue`?

Reply:
Not necessarily ... but I thought it was a nice safeguard of "never-should-happen" for `DEBUG` builds:
Since it's a `struct`, we could technically `default` it, in which case it's members will be `null`, which Nullable-Flow-Analysis does not guard against.

The other overloads don't take that struct, but reference types directly, for which Nullable-Flow-Analysis will create a warning-as-error for us when passing a nullable reference type.

I added a `message` to clarify the intent.